packages/select-menu/src/index.tsx

```diff
@@ -0,0 +1,208 @@
+  }
+  determineChildren = (appearance: SelectMenuAppearanceType) => {
+    const {children, value} = this.props
+    if (typeof children === 'function') {
+      return popoverChildrenProps =>
+        (children as SelectMenuChildrenFn<T>)({
+          ...popoverChildrenProps,
+          selectedItems: value || []
+        })
+    }
+    return this.getDefaultButton(appearance)
+  }
+  getValue = (option: T) => {
+    return option.value
+  }
+  getLabel = (option: T) => {
+    return option.label
+  }
+  getCompareBy = () => {
+    return this.props.compareBy || 'value'
+  }
+  optionClicked = (option: T) => {
+    if (this.isOptionSelected(option.value) && this.props.isMultiSelect) {
+      this.props.onDeselect(option.value)
+    } else {
+      this.props.onSelect(option.value)
+    }
+  }
+  isOptionSelected = (option: string) => {
+    if (!this.props.isMultiSelect) {
+      return this.props.value === option
+    }
+    return (this.props.value as string[]).indexOf(option) > -1
+  }
+  renderCustomItem = (option: T) => {
+    return this.props.renderItem(
+      option,
+      () => this.optionClicked(option),
+      this.isOptionSelected(option.value),
+      this.props.options
+    )
+  }
+  changeFilter = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      currentFilter: e.target.value
+    })
+  }
+  render() {
+    return (
+      <Popover
+        minHeight={this.props.height}
+        minWidth={this.props.width}
+        content={({close}) => {
+          return (
+            <S.PopoverHost>
+              {this.props.hasTitle && (
+                <S.PopoverHeader>
+                  <Strong color="intense" variant={400}>
+                    {this.props.title}
+                  </Strong>
+                  <S.CloseButton
+                    appearance="minimal"
+                    height={24}
+                    onClick={close}
+                  >
+                    <svg viewBox="0 0 16 16">
+                      <path
+                        d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L9.41 8z"
+                        fillRule="evenodd"
+                      />
+                    </svg>
+                  </S.CloseButton>
+                </S.PopoverHeader>
+              )}
+              {this.props.hasFilter && (
+                <S.FilterHost>
+                  <S.FilterInput
+                    appearance="underline"
+                    value={this.state.currentFilter}
+                    onChange={this.changeFilter}
+                    placeholder="Filter..."
+                  />
+                </S.FilterHost>
+              )}
+              <S.OptionHost appearance={this.props.appearance}>
+                {this.getFilteredOptions().map(option => {
```

Comment:
handle scenario where no filter results are available, some placeholder text ("No items found", preferably configurable) should appear